src/rules/mixin-no-risky-parent-selectors/index.js

```diff
@@ -0,0 +1,64 @@
+"use strict";
+
+const { utils } = require("stylelint");
+const namespace = require("../../utils/namespace");
+const ruleUrl = require("../../utils/ruleUrl");
+
+const ruleName = namespace("mixin-no-risky-parent-selectors");
+
+const messages = utils.ruleMessages(ruleName, {
+  rejected: `Unexpected nested parent selector in @mixin rule.`
+});
+
+const meta = {
+  url: ruleUrl(ruleName)
+};
+
+function isWithinMixin(node) {
+  let parent = node.parent;
+  while (parent) {
+    if (parent.type === "atrule" && parent.name === "mixin") {
+      return true;
+    }
+    parent = parent.parent;
+  }
+  return false;
+}
+
+function hasNestedParentSelector(selectors) {
+  return selectors
+    .split(",")
+    .some(selector => selector.includes("&") && /(.)+&/.test(selector));
```

Comment:
I think this incorrectly will flag `.foo, &` as a nested selector because of the space before `&`. (Also, the parentheses in `(.)+&` are unnecessary—since you aren't using capture groups, it behaves the same as `.+&`).

Reply:
Right, I actually had `selector.replace(' ','')` to remove spaces, not sure why I removed that.